kubernetes/kubernetes.go

```diff
@@ -443,11 +490,17 @@ func (ka KubeAgentConfig) sendMetrics(metricSample *os.File) {
 	}
 }
 
-func (ka KubeAgentConfig) sendMetricsBasedOnUploadMode(customS3Mode bool, metricSample *os.File) {
-	if customS3Mode {
+// sendMetricsBasedOnUploadMode checks the agentProvider for a custom upload config. If none are found it will use the
+// default upload mode. This will allow for more custom provider upload options in the future.
+func (ka KubeAgentConfig) sendMetricsBasedOnUploadMode(agentProvider string, metricSample *os.File) {
+	switch agentProvider {
+	case "s3":
```

Comment:
Looks like this will break custom s3 uploads. Seems like `agentProvider` is `aws` now currently, should swap for a const or enum on upload types.